### PR TITLE
Add y-flow reroute events to Reroute topology.

### PR DESCRIPTION
**Pull request by dpoltavets**
**MERGE NOTE:** merge after #4549 and change the base correspondingly.

### CODE REVIEW

File: src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteService.java [hunk 1]
```diff
@@ -105,16 +111,25 @@ public void rerouteAffectedFlows(MessageSender sender, String correlationId, Rer
...
                     flowStatusInfo = command.getReason();
                 }
                 flowRepository.updateStatusSafe(flow, flowStatus, flowStatusInfo);
 
-                if (flowPathFound) {
-                    result.flowsForReroute.add(flow);
+                if (rerouteRequired) {
+                    if (flow.getYFlowId() != null) {
+                        try {
+                            result.yFlowsForReroute.add(getYFlow(flow.getYFlowId()));
```
Comment (sergii-iakovenko) [inline src-java/reroute-topology/reroute-storm-topology/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteService.java:125]: I'd consider to implement flow.getYFlow() and directly get the y-flow.

**Review by sergii-iakovenko — APPROVED** [review]
Looks good. Minor comment left.

**Review by niksv — APPROVED** [review]

File: src-java/kilda-model/src/main/java/org/openkilda/model/Flow.java [hunk 2]
```diff
@@ -106,7 +106,7 @@ public Flow(@NonNull String flowId, @NonNull Switch srcSwitch, @NonNull Switch d
                 .priority(priority).pinned(pinned).pathComputationStrategy(pathComputationStrategy)
                 .targetPathComputationStrategy(targetPathComputationStrategy)
                 .loopSwitchId(loopSwitchId).affinityGroupId(affinityGroupId)
-                .yFlowId(yFlowId);
+                .yFlowId(yFlowId).yFlow(yFlow);
```
Comment (sergii-iakovenko) [inline src-java/kilda-model/src/main/java/org/openkilda/model/Flow.java:109]: What is the purpose to include YFlow entity in equal and hashcode calculations? yFlowId is enough.

File: src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/frames/FlowFrame.java [hunk 3]
```diff
@@ -531,6 +532,19 @@ public String getYFlowId() {
         return subFlowFrames.get(0).getYFlowId();
     }
 
+    @Override
+    public YFlow getYFlow() {
```
Comment (sergii-iakovenko) [inline src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/frames/FlowFrame.java:536]: Does it make sense to return Optional?

File: src-java/kilda-model/src/main/java/org/openkilda/model/Flow.java (hunk 2 above)
Comment (dpoltavets) [reply src-java/kilda-model/src/main/java/org/openkilda/model/Flow.java:109]: YFlow entry has been removed from the equal and hashcode calculations.

File: src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/frames/FlowFrame.java (hunk 3 above)
Comment (dpoltavets) [reply src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/frames/FlowFrame.java:536]: I do not get the point of it.

**Pull request merged**